### ISSUE TITLE
How can I tell pyenv to install the newest Python version from a given Major and Min version? aka `pyenv install 3.9.*`

### ISSUE THREAD

**Issue by John15321** [pyenv/pyenv#2407]
### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): macOS
- [x] OS architecture (e.g. amd64): aarch64
- [x] pyenv version: `pyenv 2.3.2`
- [x] Python version: `3.9.11`
- [x] C Compiler information (e.g. gcc 7.3):  `Apple clang version 13.1.6 (clang-1316.0.21.2.5)`

I would like pyenv to install the latest version of a given minor release. For example lets say I want to install Python `3.9` so naturally I would like to install the latest one with all the bug fixes etc. When listing those versions in `pyenv` I get this:

```
$ pyenv install --list
    ...
  3.9.0
  3.9-dev
  3.9.1
  3.9.2
  3.9.4
  3.9.5
  3.9.6
  3.9.7
  3.9.8
  3.9.9
  3.9.10
  3.9.11
  3.9.12
  3.9.13
    ...
```

Which is fine, but now how can I tell pyenv to simply install the latest one, in this case, that would be `3.9.13` without saying that explicitly? I know that syntax like this does not work `pyenv install 3.9.*`. But is there another way to do that?

**Comment by native-api** [comment]
`pyenv install 3.9:latest`

**Comment by John15321** [comment]
> `pyenv install 3.9:latest`

Honestly, I couldn't find it anywhere. Is this option shown somewhere in the documentation or in the `--help`?

**Comment by native-api** [comment]
> > `pyenv install 3.9:latest`
> 
> Honestly, I couldn't find it anywhere. Is this option shown somewhere in the documentation or in the `--help`?

It's specified in https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-install

It's not present in the `--help` text -- that's probably an oversight.

**Comment by John15321** [comment]
> > > `pyenv install 3.9:latest`
> > 
> > 
> > Honestly, I couldn't find it anywhere. Is this option shown somewhere in the documentation or in the `--help`?
> 
> It's specified in https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-install
> 
> It's not present in the `--help` text -- that's probably an oversight.

I think it would be very good to know for any user at the beginning that they can use it in this way. The `3.9.x` `x` is a bug fix so it does not impact the in which the current python distribution should work. I would suggest adding it in `pyenv install --help` or something

**Comment by brunns** [comment]
Is there an analogue of this for `pyenv local` and `pyenv global`? Having installed the latest version by Major & Minor, without knowing the Patch version, how do I then use it?

**Comment by John15321** [comment]
> Is there an analogue of this for `pyenv local` and `pyenv global`? Having installed the latest version by Major & Minor, without knowing the Patch version, how do I then use it?

Yeah I would also like to know

**Comment by native-api** [comment]
Implemented auto-resolving micro versions in https://github.com/pyenv/pyenv/pull/2487

**Comment by John15321** [comment]
> Implemented auto-resolving micro versions in #2487

Im sorry but could you provide an example usage?

**Comment by native-api** [comment]
https://github.com/pyenv/pyenv#prefix-auto-resolution
